docs/api.md

```diff
@@ -0,0 +1,195 @@
+# Documentation for Scouting Backend HTTP API
+
+## Authenticated Requests
+
+Authenticated requests should have a Authentication header with the format "Authentication: Bearer {signed jwt string}".
+
+Almost all request that have to deal with users or are not a GET request are authenticated with few exceptions.
+
+---
+
+## /authenticate - POST
+
+For retrieving a JWT token for authenticated requests.
+
+### Request Body
+
+```json
+{
+  "username": "frank",
+  "password": "asdf"
+}
+```
+
+### Response Body
+
+```json
+{
+  "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1MTMyMTA0OTksInN1YiI6ImZyYW5rIn0.HCwmwj0f-4b2I-hK9QEJ-4berztETH_IDgcSIJBPXMI"
+}
+```
+
+---
+
+## /users - GET - Authenticated
+
+Gets all users.
+
+### Response Body
+
+```json
+[
+  {
+    "username": "frank",
+    "hashedPassword": "$2b$12$29H98PpF9E/CccCc.OEBfObX4Sm0coLZE6cCWSAmaNQkveGhop5lW"
+  },
+  ...
+]
+```
+
+---
+
+## /users - POST - Authenticated
+
+Creates a new user.
+
+### Request Body
+
+```json
+{
+  "username": "frank2",
+  "password": "asdf"
+}
+```
+
+---
+
+## /users/{username} - GET - Authenticated
+
+Gets a user.
+
+### Response Body
+
+```json
+{
+  "username": "frank",
+  "hashedPassword": "$2b$12$29H98PpF9E/CccCc.OEBfObX4Sm0coLZE6cCWSAmaNQkveGhop5lW"
+}
+```
+
+---
+
+## /users/{username} - DELETE - Authenticated
+
+Deletes a user.
+
+---
+
+## /events - GET
+
+Gets all (basic) events.
+
+### Response Body
+
+```json
+[
+  {
+    "key": "2017alhu",
+    "name": "Rocket City Regional",
+    "shortName": "Rocket City",
+    "date": "2017-03-21T17:00:00-07:00"
```

Comment:
Not sure if this matters, but some of these times are in different time zones. Should be fine though, they're all the same format.